woocommerce/woocommerce-ios
# Shipping Labels M2: display the Package List

Closes #3958  

## Description
In this PR I implemented the screen for displaying the Package List in the Shipping Labels.
All the components are implemented in SwiftUI. The flow was implemented using a view that contains another subview. Wanting to be more specific, `ShippingLabelAddNewPackage` is the container of the reusable `ShippingLabelPackageList` view. I implemented it in this way because in the future we will use the `ShippingLabelPackageList` component for the Package Creation Flow. So, the `ShippingLabelPackageList` will be embedded in another view where the user can create a Custom Package or select an existing package.

## Changes
- Implemented the `ShippingLabelPackageList` view that displays a list of Custom Packages and Predefined Packages. I also implemented the view model `ShippingLabelPackageListViewModel` and the tests. The view model takes care of managing the state of the view and the selected package.
- I made some small UI changes to the existing `ShippingLabelAddNewPackage` view, removing the unused (for the moment) `SegmentedView`. I also made some changes to the view model, which now takes care of the States and of fetching the package details. I added the unit tests also for this view model.
- I implemented a new reusable view in SwiftUI, called `SelectableItemRow` that displays a title, a subtitle, and the checkmark on the left side when selected.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the new screen that shows the Package Details screen and all the items that will be shipped.
8. Press the "Package selected" row. -> You should see the new screen with the list of the existing packages, with Custom Packages on top if any.

For the moment the `error` and the `syncing` states are not managed and will be part of other PRs. https://github.com/woocommerce/woocommerce-ios/issues/3965 https://github.com/woocommerce/woocommerce-ios/issues/3966 Also the handle of the value will be part of another PR https://github.com/woocommerce/woocommerce-ios/issues/3964

| Package List (Light mode)             |  Package List (Dark mode)  |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 12 Pro - 2021-04-15 at 14 26 30](https://user-images.githubusercontent.com/495617/114868993-cbb3b580-9df6-11eb-8b1d-315731731573.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-04-15 at 14 27 19](https://user-images.githubusercontent.com/495617/114869003-ce160f80-9df6-11eb-8397-8a2195f3e391.png)



Update release notes:
